### PR TITLE
AV-54052: fix package-lock

### PR DESCRIPTION
**Pull request by osfameron**
Looks like we need to have site-generator at 3.1, for `npx antora` to Just Work.
Unsure if we should now also remove the site-generator-ms entry? Ping @mojavelinux

### CODE REVIEW

**Comment by mojavelinux** [comment]
Adding the site generator package shouldn't be necessary as it is already pulled in by the antora package.

Yes, you still need site-generator-ms because it's not a site generator anymore, but rather a collection of Antora extensions which are still being used. The goal is to eventually be able to use only packages from the Antora project (such as Antora Atlas and Antora Navigator), but those packages are not developed enough to replace the extensions in site-generator-ms.

Since this build is using a package.json file, using `npx` still requires running `npm ci` first. When `npx` is run without a node_modules directory, it tries to only use the packages that are specified in the command, which are not enough to build this site.

**Comment by osfameron** [comment]
Ah! This is because `npx ci` gives a cleaner install than `npm i` ?

Having done the latter, my colleagues were finding that `npx antora -v` gave the expected 3.1.x, while `npx antora some-playbook` gave the `FATAL (antora): Generator not found or failed to load. Try installing the '@antora/site-generator' package.` message.

**Comment by mojavelinux** [comment]
> This is because npx ci gives a cleaner install than npm i ?

`ci` installs dependencies listed in package-lock.json so that every installs exactly the same versions. Only the person managing this file needs to run `i`, which is what generates this file.

See:

* https://docs.npmjs.com/cli/v9/commands/npm-ci
* https://docs.npmjs.com/cli/v9/commands/npm-install

**Comment by osfameron** [comment]
Ah! Thanks, that's been a mystery... I need to learn Node.js properly...
Looks like we accidentally downgraded the lockfile last year. I'll keep an eye on this from now on 👍

**Comment by osfameron** [comment]
Nice. Thanks, all done, will merge after review from colleagues, and read up on Node/npm...

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fund=false
+lockfile-version=3
```
Comment (osfameron) [inline .npmrc:2]: This is the key line: we pin to version 3

Comment (osfameron) [reply .npmrc:2]: Reviewing package-lock.json is tricky (Github can't cope with it... command-line git is fine)
You could note that:

line 3: `"lockfileVersion": 3,`
line 186:
```
    "node_modules/@antora/site-generator": {
      "version": "3.1.4",
```
so even though we've not, in the end, added site-generator explicitly, it is pulled in by the version 3 lockfile.

Comment (osfameron) [reply .npmrc:2]: For bonus points, check out this PR, and then

```
npm ci
npx antora -v
```
and check that both cli and site-generator are 3.1.4

**Pull request merged**